File: geekbang/algorithm/source-code/1/lc88 合并两个有序数组.go
func merge(nums1 []int, m int, nums2 []int, n int) {
	// 转换思路，从末尾开始往前
	index := len(nums1) - 1
	// 分别从两个输入数组的末尾开始遍历
	m = m - 1
	n = n - 1
	for m >= 0 && n >= 0 {
		num1 := nums1[m]
		num2 := nums2[n]
		if num1 > num2 {
			nums1[index] = num1
			m--
		} else {
			nums1[index] = num2
			n--
		}
		index--
	}
	// 如果最后nums2有剩余，直接拷贝到nums1即可
	if n >= 0 {
		copy(nums1[0:n+1], nums2[0:n+1])
	}
}

File: geekbang/algorithm/source-code/1/lc88 合并两个有序数组.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    let i = m - 1;
    let j = n - 1;
    for (let k = m + n - 1; k >= 0; k--) {
        // 什么是要nums1[i]？或者j出界了，或者i,j都没出界，要大的
        if (j < 0 || (i >= 0 && nums1[i] >= nums2[j])) {
            nums1[k] = nums1[i];
            i--;
        } else {
            nums1[k] = nums2[j];
            j--;
        }
    }
};